### PR TITLE
check key and cert are valid before sending to Envoy

### PR DESCRIPTION
**Pull request by ramaraochavali**
Currently we only watch cert and assume when cert changes key is also changed (wait for some debounce duration) and send it to Envoy. We have seen in some cases cert changes, but key is not changed with in the debounce period, leading to sending incorrect cert and key combination to Envoy resulting in cluster/listener NACKs.

This PR validates that key and cert form a correct combination with a retry (starting with debounce , upto max 1min). If the  key and cert are not a valid combination it returns error - to avoid NACKs.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [X] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: security/pkg/nodeagent/cache/secretcache_test.go [hunk 1]
```diff
@@ -377,21 +377,24 @@ func runFileAgentTest(t *testing.T, sds bool) {
 	if err := file.AtomicWrite(sc.existingCertificateFile.CertificatePath, testcerts.RotatedCert, os.FileMode(0o644)); err != nil {
 		t.Fatal(err)
 	}
+	if err := file.AtomicWrite(sc.existingCertificateFile.PrivateKeyPath, testcerts.RotatedKey, os.FileMode(0o644)); err != nil {
```
Comment (ramaraochavali) [inline security/pkg/nodeagent/cache/secretcache_test.go:380]: The new code caught this error :-)

**Review by howardjohn — APPROVED** [review]

**Pull request merged**